src/Octoshift/SecretScanningAlertService.cs

```diff
@@ -0,0 +1,169 @@
+            $"Migrating Secret Scanning Alerts from '{sourceOrg}/{sourceRepo}' to '{targetOrg}/{targetRepo}'");
+
+        var sourceAlerts = await GetAlertsWithLocations(_sourceGithubApi, sourceOrg, sourceRepo);
+        var targetAlerts = await GetAlertsWithLocations(_targetGithubApi, targetOrg, targetRepo);
+
+        _log.LogInformation($"Source {sourceOrg}/{sourceRepo} secret alerts found: {sourceAlerts.Count}");
+        _log.LogInformation($"Target {targetOrg}/{targetRepo} secret alerts found: {targetAlerts.Count}");
+
+        _log.LogInformation("Matching secret resolutions from source to target repository");
+        foreach (var alert in sourceAlerts)
+        {
+            _log.LogInformation($"Processing source secret {alert.Alert.Number}");
+
+            if (SecretScanningAlerts.IsOpen(alert.Alert.State))
+            {
+                _log.LogSuccess("  secret alert is still open, nothing to do");
+            }
+            else
+            {
+                _log.LogInformation("  secret is resolved, looking for matching detection in target...");
+                var target = MatchTargetSecret(alert, targetAlerts);
+
+                if (target == null)
+                {
+                    _log.LogWarning(
+                        $"  failed to locate a matching secret to source secret {alert.Alert.Number} in {targetOrg}/{targetRepo}");
+                }
+                else
+                {
+                    _log.LogInformation(
+                        $"  source secret alert matched alert to {target.Alert.Number} in {targetOrg}/{targetRepo}.");
+
+                    if (alert.Alert.Resolution == target.Alert.Resolution
+                        && alert.Alert.State == target.Alert.State)
+                    {
+                        _log.LogSuccess("  source and target alerts are already aligned.");
+                    }
+                    else
+                    {
+                        _log.LogInformation(
+                            $"  updating target alert:{target.Alert.Number} to state:{alert.Alert.State} and resolution:{alert.Alert.Resolution}");
+
+                        if (dryRun)
+                        {
+                            _log.LogInformation(
+                                $"  executing in dry run mode! Secret alert, {target.Alert.Number}, in repository {targetOrg}/{targetRepo} would have been updated to resolution, {alert.Alert.Resolution}");
+                        }
+                        else
+                        {
+                            await _targetGithubApi.UpdateSecretScanningAlert(targetOrg, targetRepo, target.Alert.Number,
+                                alert.Alert.State, alert.Alert.Resolution);
+                            _log.LogSuccess(
+                                $"  source and target alert state and resolution have been aligned to {alert.Alert.Resolution}.");
+                        }
+                    }
+                }   
```

Comment:
there is too much `if-else` here and it makes it hard to follow the code. The indentations can be greatly reduced by checking for a condition and `continue`, for example:
```c#
if (SecretScanningAlerts.IsOpen(alert.Alert.State))
{
    _log.LogSuccess(" secret alert is still open, nothing to do");
    continue:
}

_log.LogInformation(" secret is resolved, looking for matching detection in target...");

...

if (target is null)
{
    _log.LogWarning(...);
    continue;
}

...
```

Reply:
Acknowledged and will refactor